core/src/test/java/com/google/errorprone/bugpatterns/BugCheckerTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2023 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.bugpatterns;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.CompilationTestHelper;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.VariableTree;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BugCheckerTest {
+  @Test
+  public void isSuppressed_withoutVisitorState() {
+    CompilationTestHelper.newInstance(LegacySuppressionCheck.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "class A {",
+            "  void m() {",
+            "    // BUG: Diagnostic contains: []",
+            "    int unsuppressed;",
+            "    // BUG: Diagnostic contains: []",
+            "    @SuppressWarnings(\"foo\") int unrelatedSuppression;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings(\"Suppressible\") int suppressed;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings(\"Alternative\") int suppressedWithAlternativeName;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings(\"all\") int allSuppressed;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings({\"foo\", \"Suppressible\"}) int alsoSuppressed;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings({\"all\", \"foo\"}) int redundantlySuppressed;",
+            "    // BUG: Diagnostic contains: [Suppressible]",
+            "    @SuppressWarnings({\"all\", \"OnlySuppressedInsideDeprecatedCode\"}) int ineffectiveSuppression;",
+            "    // BUG: Diagnostic contains: []",
+            "    @Deprecated int unuspportedSuppression;",
+            "  }",
+            "}")
+        .doTest();
+  }
```

Comment:
The final test case here shows why the code under test is deprecated: the `suppressionAnnotations = Deprecated.class` directive isn't respected.